app/src/main/java/berlin/mfn/naturblick/ui/composable/ToggleGPSFab.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2024 Museum für Naturkunde Berlin.
+ * This code is licensed under MIT license (see LICENSE.txt for details)
+ */
+
+package berlin.mfn.naturblick.ui.composable
+
+import android.Manifest
+import android.content.pm.PackageManager
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.core.content.ContextCompat
+import berlin.mfn.naturblick.R
+
+@Composable
+fun ToggleGPSFab(
+    locationEnabled: Boolean,
+    modifier: Modifier = Modifier,
+    onToggle: (Boolean) -> Unit
+) {
+    val permissionRequest = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestMultiplePermissions(),
+        onResult = { permissions ->
+            if (permissions.values.any { it }) {
```

Comment:
Are you sure that `any` is correct? I would think that `all` is correct.

Reply:
i see, you are asking for `COARSE` respectively `FINE` location and one is sufficient.